### DISCUSSION TITLE
NextJS server-side debug won't start using VS Code since upgrading to version 14

### DISCUSSION THREAD

**Discussion by olivierr91** [vercel/next.js#59554] [Help]
### Summary

Since upgrading to Next.js version 14.0.4 from version 12, I cannot start server-side debugging in VS Code on Windows. Everything works when I manually run `npm run dev`, or `npm run build` && `npm run start`. But when I click on "Next.js: debug server-side" in the VS Code "Run and Debug" tab, I only see the following output in terminal:

```
Debugger attached.

> dev
> next dev
```

The server never starts. I should be seeing some output about the next.js version and the server being started like when I run `npm run dev`. When I go to the `http://localhost:3000` in the browser, I get an ERR_CONNECTION_REFUSED. I don't know if this is related, but I can see this exception when I select to break on all exceptions:

[![enter image description here][1]][1]

Here is my launch.json file:

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Next.js: debug server-side",
            "type": "node-terminal",
            "request": "launch",
            "cwd": "${workspaceFolder}\\src",
            "command": "npm run dev"
        },
        {
            "name": "Next.js: debug client-side",
            "type": "chrome",
            "request": "launch",
            "cwd": "${workspaceFolder}\\src",
            "url": "http://localhost:3000"
        },
        {
            "name": "Next.js: debug full stack",
            "type": "node-terminal",
            "request": "launch",
            "cwd": "${workspaceFolder}\\src",
            "command": "npm run dev",
            "serverReadyAction": {
                "pattern": "started server on .+, url: (https?://.+)",
                "uriFormat": "%s",
                "action": "debugWithChrome"
            }
        }
    ]
}
```

And here's my `package.json` section:

```
{
    "scripts": {
        "dev": "next dev",
        "build": "next build",
        "export": "next build && next export",
        "start": "next start",
        "lint": "next lint",
        "test": "jest --watch",
        "storybook": "storybook dev -p 6006",
        "build-storybook": "storybook build"
    },
    "dependencies": {
        "classnames": "^2.3.1",
        "js-cookie": "^3.0.1",
        "next": "^14.0.4",
        "next-intl": "^3.0.1",
        "react": "^18.2.0",
        "react-animate-height": "^3.0.5",
        "react-dom": "^18.2.0",
        "react-gtm-module": "^2.0.11",
        "react-svg": "^16.1.31",
        "reflect-metadata": "^0.1.13",
        "tsyringe": "^4.8.0",
        "uuid": "^9.0.1"
    },
    "devDependencies": {
        "@storybook/addon-essentials": "^7.5.3",
        "@storybook/addon-interactions": "^7.5.3",
        "@storybook/addon-links": "^7.5.3",
        "@storybook/addon-onboarding": "^1.0.8",
        "@storybook/blocks": "^7.5.3",
        "@storybook/nextjs": "^7.5.3",
        "@storybook/react": "^7.5.3",
        "@storybook/testing-library": "^0.2.2",
        "@testing-library/jest-dom": "^6.1.4",
        "@testing-library/react": "^14.0.0",
        "@types/jest": "^29.5.6",
        "@types/js-cookie": "^3.0.2",
        "@types/node": "^18.7.1",
        "@types/react": "^18.0.17",
        "@types/react-dom": "^18.0.6",
        "@types/react-gtm-module": "^2.0.2",
        "eslint": "^8.51.0",
        "eslint-config-next": "^13.5.6",
        "jest": "^29.7.0",
        "jest-environment-jsdom": "^29.7.0",
        "sass": "^1.54.5",
        "storybook": "^7.4.6",
        "typescript": "^5.2.2"
    }
}
```

  [1]: https://i.stack.imgur.com/4wepF.png

### Additional information

```js
Windows 11 x64
```

**Comment by hnykda** [comment]
Same here, don't know what to do...

**Comment by Burov96** [comment]
same here too!

**Comment by lyonb96** [comment]
I found that if you click the "continue" button on the debug tool in the VS code window, it will finish starting the dev server:
![image](https://github.com/vercel/next.js/assets/31573074/a756ecb7-138d-4d45-8985-7b33b4a95278)
I couldn't find a way to prevent it from hitting this error, and occasionally other errors would pop up after clicking this button, but eventually just continuing to click that button would get the server to start. Obviously this isn't an ideal situation, but this is a workaround for now. If anyone has any tips for preventing it from catching on this, I'd love to hear it!

Edit: unchecking "All Exceptions" in the debugger tab of VS Code got me around the issue - the dev server now launches without a hitch.

**Comment by NghiaHuynh** [comment]
The same issue run on mac osx Monterey but my windown still start server-side debugging in VS Code
